articles/working-with-forms-in-django/index.md

```diff
@@ -0,0 +1,217 @@
+```
+
+Once this is done successfully, try creating a new user again. **Bravo! It worked.** So, a new user account is created, validated and the user is redirected to the home page after creating an account.
+
+**Note:** This form does not have an email field provided for us. Therefore, we'll need to create an email field but remember Django already provides this form for us through its UserCreationForm. So in a bit, we'll be looking at how to edit the Django form to include `emailField` and any other field necessary.
+
+To do this, we'll start by creating a new file **forms.py**
+
+```python
+from django import forms
+from django.contrib.auth.models import User
+from django.contrib.auth.forms import UserCreationForm
+
+class UserRegisterForm(UserCreationForm):
+    email = forms.EmailField()
+
+    class Meta:
+        model = User
+        fields = ['username', 'email', 'password1', 'password2']
+```
+
+*Class Meta* is used in adding fields in an already existing Django `ModelForm.` There are various Django fields. [Refer documentation](https://docs.djangoproject.com/en/3.1/topics/forms/modelforms/#field-types) for more details on Django forms.
+
+In **views.py:**
+
+```python
+from django.shortcuts import render, redirect
+from .forms import UserRegisterForm
+
+def register(request):
+    if request.method == 'POST':
+      form = UserRegisterForm(request.POST)
+      if form.is_valid(): 
+        form.save()
+        username = form.cleaned_data.get('username')
+        return redirect('home')
+    else:
+      form = UserRegisterForm()
+  return render(request, 'details/register.html', {'form': form})
+```
+
+Note how we changed from UserCreationForm to UserRegisterForm.
+
+![Register form 2 with emailField]()
+
+
+A few additional styling can be done using either CSS or bootstrap within the template. This additional styling is done using ***Django-crispy-forms**.*  This is an application that allows Django-forms to be styled without re-writing the codes.
+
+Swiftly, let's add crispy-forms. Firstly, install crispy forms in your terminal.
+
+`pip install Django-crispy-forms`
+
+Once this is installed, go to your **settings.py** file and add the crispy form under installed_apps
+
+![Settings.py file]()
+
+
+Then, at the bottom of the **settings.py** file, add these lines of code.
+
+```python
+CRISPY_TEMPLATE_PACK = 'bootstrap4'
+```
+
+Lastly, in your **register.html** file. Add the crispy form tag at the top, this way:
+
+```python
+{% load crispy_forms_tags %}
+<h1>Register Form</h1>
+<form method="POST">
+  {% csrf_token %} 
+  <fieldset class="form-group">
+    {{ form|crispy }}
+  </fieldset>
+  <div class="form-group">
+    <button class="btn btn-outline-info" type="submit">Register</button>
+  </div>
+</form>
+```
+
+Notice, we also added "**crispy"** in our code. Hence we are adding crispy to do the editing. Don't skip that.
+
+Once this is done properly, your form would look like this.
+
+ 
+
+![Register form 4 with crispy]()

```

Comment:
I think you have missed the image file names. Can you insert the appropriate images at their respective places and LMK?

Reply:
Changes have been made. Thank you